RFCs/0011-partiql-insert-update-spec.md

```diff
@@ -0,0 +1,761 @@
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('T_601', 'Yojimbo', 106, '1961-06-16', 'Drama');
+```
+
+The following examples use the `DEFAULT` clause for the date attributes rather than specifying a value:
+
+```SQL
+INSERT INTO films
+VALUES ('UA502', 'Bananas', 105, DEFAULT, 'Comedy', '82 minutes');
+
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('T_601', 'Yojimbo', 106, DEFAULT, 'Drama');
+```
+
+The following examples statement inserts a row or item consisting entirely of default values for required attributes:
+
+```SQL
+INSERT INTO films DEFAULT VALUES;
+```
+
+The following example statement inserts multiple rows using the multi-row VALUES syntax:
+
+```SQL
+INSERT INTO films (code, title, did, date_prod, kind)
+VALUES ('B6717', 'Tampopo', 110, '1985-02-10', 'Comedy'),
+       ('HG120', 'The Dinner Game', 140, DEFAULT, 'Comedy');
+```
+
+In the following statement, we insert two values to `Music` table, `Music` table has an open schema with `Artist` and `SongTitle` as required attributes:
+
+```SQL
+INSERT INTO Music
+    <<{'Artist' : 'Acme Band', 'SongTitle' : 'PartiQL Rocks'},
+    {'Artist' : 'Emca Band', 'TitleSong' : 'PartiQL Rocks'}>>;
+```
+
+In the following statement we insert multiple person items as a bag value to `Person` table, `Person` table has an open schema with `LastName` and `DOB` as required attributes:
+
+```SQL
+INSERT INTO Person <<{'FirstName' : 'Raul',
+    'LastName' : 'Lewis',
+    'DOB' : 1963-08-19T,
+    'GovId' : 'LEWISR261LL',
+    'GovIdType' : 'Driver License',
+    },{'LastName' : 'Logan',
+    'DOB' : 1967-07-03T,
+    'Address' : '43 Stockert Hollow Road, Everett, WA, 98203'
+    },{'LastName' : 'Pena',
+    'DOB' : 1974-02-10T,
+    'GovId' : '744 849 301',
+    'GovIdType' : 'SSN',
+    'Address' : '4058 Melrose Street, Spokane Valley, WA, 99206'
+    }>>;
+```
+
+### 3.2 Query
+
+`SELECT` query expression can be used for inserting the resulting values from the query to the target table. The resulting tuples (E.g. rows) MUST have the same attribute name and type as the target schema’s required attributes.
+
+Optional attributes in target schema that have the same name in the resulting tuples from the source schema, SHOULD have the same type. If the implementing database provides type coercion rule for the types involved in source and destination, the `SELECT` MAY lead to a successful outcome, otherwise it's a `SemanticError`. See examples for more clarification.
+
+#### 3.4 Examples
+
+The following example statement inserts items from `Vehicles` table into `HeavyVehicles` table. Both tables have the same layout for required and optional attributes, therefore the `INSERT` statement can insert items using a `SELECT` query.
+
+```SQL
+-- The following `CREATE TABLE` is arbitrary since the PartiQL DDL is yet to be
+-- defined.CREATE TABLE Music
```

Comment:
newline needed.  Also why is this statement commented but not the one below?

Reply:
This is fixed in new commits.